docs/reference/validation.txt

```diff
@@ -23,3 +23,32 @@ Mongoid behaves slightly differently to Active Record when using ``#valid?``
 on already persisted data. Active Record's ``#valid?`` will run all
 validations whereas Mongoid's ``#valid?`` will only run validations on
 documents that are in memory as an optimization.
+
+
+``validates_uniqueness_of`` and ``:conditions``  Option
+=======================================================
+
+The ``:conditions`` option to ``validates_uniqueness_of`` can be used to
+provide additional conditions to add to the database query looking for
+identical documents. This option does not influence when the validation
+is executed because it is not considered when Mongoid retrieves the present
+value of the respective field from the model. Consider the following example:
+
+.. code-block:: ruby
+
+  class Band
+    include Mongoid::Document
+    
+    field :name, type: String
+    field :year, type: Integer
+    
+    validates_uniqueness_of :name, conditions: -> { where(:year.gte => 2000) }
+  end
+  
+  # OK
+  Band.create!(name: "Sun Project", year: 2000)
+  
+  # Fails validation because there is a band with the "Sun Project" name
+  # and year 2000 in the database, even though the model being created now
+  # does not have a year.
+  Band.create!(name: "Sun Project")
```

Comment:
I'm struggling to understand the real-world use case for `:conditions`; this example doesn't explain to me how it's actually useful in an app. In my 10+ years using Mongoid and making 100s of uniqueness validations, I've never used the `:conditions` option -- but I do use `:scope` very frequently.


Is this code correct?

```ruby
Band.create!(name: "Sun Project") # OK
Band.create!(name: "Sun Project") # OK
Band.create!(name: "Sun Project") # OK
Band.create!(name: "Sun Project", year: 2000) # OK
Band.create!(name: "Sun Project") # Fails!
```

If so, it seems like option is quite dangerous. I'd propose to deprecate it and if people really want it they can write a custom validator.